spec/models/exercise_spec.rb

```diff
@@ -1,10 +1,42 @@
 require 'rails_helper'
 
 RSpec.describe Exercise, type: :model do
+  let(:current_user) { FactoryBot.create(:user) }
+  let(:exercise) { FactoryBot.create(:exercise, user: current_user) }
+
+  describe 'Basic setup methods' do
+    it 'has a valid factory' do
+      expect(exercise).to be_valid
+    end
+
+    it 'persists a valid model object when given strings for title, general description, user id and a global boolean' do
+      exercise = Exercise.new(
+        title: 'Senecas Dance with Dragons',
+        general_description: 'An amazing exercise to really get your pants off.',
+        user: current_user
+      )
+    end
+  end
+
+  describe 'Classic ActiveRecord Instance Methods' do
+    it 'returns a title' do
+      expect(exercise.title).to eq('Senecas Courage Walk')
+    end
+
+    it 'returns a general description' do
+      expect(exercise.general_description).to eq('Take a stroll with Seneca and beef up your courage.')
+    end
+
+    it 'returns a global boolean' do
+      expect(exercise.global?).to eq(false)
+    end
+  end
+
+  # Henry's custom methods
   describe '#rehearsals_for_user' do
     it 'successfully fetches all rehearsals for that particular exercise for that particular user' do
       user = FactoryBot.build_stubbed(:user)
```

Comment:
Why are you using `build_stubbed` here? Isn't this testing a database query?

Reply:
Hmm... good point. I didn't know that I needed to have all objects involved persist to the database, including `user`. I thought I could just persist `exercise` and `rehearsal` since they are the directly involved queries. 

Well, now I know! Use `create` on all classes participating in a database query.